aten/src/ATen/native/AdaptiveAveragePooling3d.cpp

```diff
@@ -30,52 +30,51 @@ static void adaptive_avg_pool3d_out_frame(
     int64_t istrideT,
     int64_t istrideH,
     int64_t istrideW) {
-  int64_t d = 0;
-  at::internal::lazy_init_num_threads();
-#pragma omp parallel for private(d)
-  for (d = 0; d < sizeD; d++) {
-    /* loop over output */
-    int64_t ot, oh, ow;
-    for (ot = 0; ot < osizeT; ot++) {
-      int istartT = start_index(ot, osizeT, isizeT);
-      int iendT = end_index(ot, osizeT, isizeT);
-      int kT = iendT - istartT;
-
-      for (oh = 0; oh < osizeH; oh++) {
-        int istartH = start_index(oh, osizeH, isizeH);
-        int iendH = end_index(oh, osizeH, isizeH);
-        int kH = iendH - istartH;
-
-        for (ow = 0; ow < osizeW; ow++) {
-          int istartW = start_index(ow, osizeW, isizeW);
-          int iendW = end_index(ow, osizeW, isizeW);
-          int kW = iendW - istartW;
-
-          /* local pointers */
-          scalar_t* ip = input_p + d * istrideD + istartT * istrideT +
-              istartH * istrideH + istartW * istrideW;
-          scalar_t* op = output_p + d * osizeT * osizeH * osizeW +
-              ot * osizeH * osizeW + oh * osizeW + ow;
-
-          /* compute local average: */
-          scalar_t sum = 0;
-          int it, ih, iw;
-          for (it = 0; it < kT; it++) {
-            for (ih = 0; ih < kH; ih++) {
-              for (iw = 0; iw < kW; iw++) {
-                scalar_t val =
-                    *(ip + it * istrideT + ih * istrideH + iw * istrideW);
-                sum += val;
+  at::parallel_for(0, sizeD, 1, [&](int64_t start, int64_t end) {
```

Comment:
this parallel region is nested inside another one, I suspect it won't actually be parallelized, can you please check?
Good fix otherwise, we should get rid of #pragma omps. Can you please check if there are any other #pragma omps and get rid of them too?

Reply:
Yes,  it is nested for dim=5, but for dim=4, it will be parallelized. I will check if there are any other #pragma omps.

Reply:
https://github.com/pytorch/pytorch/blob/508bab43e71154b894cae04a6fa88547fd3c1505/aten/src/ATen/native/MaxUnpooling.cpp#L42 use #pragma omps, but there has a **#omp critical**, which needs refact the code to remove it(may do big change).  so merge this PR at first.